hilti/runtime/include/types/interval.h

```diff
@@ -36,8 +37,18 @@ class Interval {
      * Constructs an interval from a double value.
      *
      * @param secs interval in seconds.
+     * @throws OutOfRange if *secs* cannot be represented with the internal resolution
      */
-    explicit Interval(double secs, SecondTag /*unused*/) : _nsecs(static_cast<int64_t>(secs * 1e9)) {}
+    explicit Interval(double secs, SecondTag /*unused*/)
+        : _nsecs([&]() {
+              auto x = secs * 1'000'000'000;
+
+              auto limits = std::numeric_limits<int64_t>();
```

Comment:
Might as well declare this as `constexpr` so we have a chance of generating less work at runtime.
```.cpp
               constexpr auto limits = std::numeric_limits<int64_t>(); 
```

Reply:
ack